config-model/src/main/javacc/SchemaParser.jj

```diff
@@ -2439,17 +2442,40 @@ void fieldRankType(ParsedRankProfile profile) :
 void fieldRankFilter(ParsedRankProfile profile) :
 {
     String name;
-    double filterThreshold;
 }
 {
     <RANK> name = identifier()
     ( ( <COLON> <FILTER> ) { profile.addFieldRankFilter(name, true); }
-    | ( lbrace() <FILTER_THRESHOLD> <COLON> filterThreshold = floatValue()
-        { profile.addFieldRankFilterThreshold(name, filterThreshold); }
-        ( <NL> )* <RBRACE> )
+    | ( lbrace() ( fieldRankItem(profile, name) ( <NL> )* )* <RBRACE> )
     )
 }
 
+void fieldRankItem(ParsedRankProfile profile, String field) :
+{
+    double filterThreshold;
+    ElementGap elementGap;
+}
+{
+    (   <FILTER_THRESHOLD> <COLON> filterThreshold = floatValue()
+        { profile.addFieldRankFilterThreshold(field, filterThreshold); } |
```

Comment:
For grammars I think we generally prefer to have the rule `|` operator left-aligned to make it more clearly visible that these are different matching paths.

I.e. something like
```
(   <FILTER_THRESHOLD> <COLON> filterThreshold = floatValue()
        { profile.addFieldRankFilterThreshold(field, filterThreshold); }
|   <ELEMENT_GAP> <COLON> elementGap = elementGapValue()
        { profile.addFieldRankElementGap(field, elementGap); }
)
```
Though I see this pattern is not consistently applied...